W-16759088-editing-Funtions&Formats-2.8-dm (#389)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-formats-flatfile.adoc

|`schemaPath` (Required) |`String`|`null`|Path to the schema definition. Specifies the location in your local disk of the schema file that parses your input.
|`segmentIdent`  |`String`|`null`|Segment identifier in the schema for fixed-width or COBOL copybook schemas. Required when parsing a single segment or record definition if the schema includes multiple segment definitions.
|`structureIdent`  |`String`|`null`|Structure identifier in the schema for flat file schemas. Required when parsing a structure definition if the schema includes multiple structure definitions.
|`substituteCharacterAsMissingValue`  |`Boolean`|`false`|If set to `true` and if the `missingValues` property is set to `none` and the `useMissCharAsDefaultForFill` is set to `true`, the flat file parser will handle the `0x1A` (SUB) character as the missing value. +

> Otherwise, the flat file parser will not handle the `0x1A` character as the missing value.

Author: ```suggestion
|`substituteCharacterAsMissingValue`  |`Boolean`|`false`|If set to `true` and the `missingValues` property is set to `none`, with `useMissCharAsDefaultForFill` set to `true`, the flat file parser handles the `0x1A` (SUB) character as the missing value. +
Otherwise, the flat file parser doesn't handle the `0x1A` character as the missing value.
```

Valid values are `true` or `false`.
|`truncateDependingOn`  |`Boolean`|`false`|For COBOL copybook, truncates DEPENDING ON values to the length used.

Valid values are `true` or `false`.

> |`useMissCharAsDefaultForFill`  |`Boolean`|`false`|By default, the flat file reader and writer use spaces for missing characters and ignore the setting of the `missingValues` property. When you set this property to `true`, DataWeave honors the setting of the `missingValues property`. Introduced in DataWeave 4.3 (2.3.0-20210823) for the September 2021 release of Mule 4.3.0-20210823.

Author: ```suggestion
|`useMissCharAsDefaultForFill`  |`Boolean`|`false`|By default, the flat file reader and writer use spaces for missing characters and ignore the setting of the `missingValues` property. When you set this property to `true`, DataWeave honors the `missingValues property` setting. Introduced in DataWeave 2.3 (2.3.0-20210823) for the September 2021 release of Mule 4.3.0-20210823.
```

Valid values are `true` or `false`.
|`recordTerminator`  |`String`|`null`|Line break for a record separator. DataWeave uses this property as a separator only when there are multiple records. Values translate directly to character codes, and `none` leaves no termination on each record.

> 

Author: ```suggestion
```

|`truncateDependingOn`  |`Boolean`|`false`|For COBOL copybook, truncates DEPENDING ON values to the length used.

Valid values are `true` or `false`.

> |`useMissCharAsDefaultForFill`  |`Boolean`|`false`|By default, the flat file reader and writer use spaces for missing characters and ignore the setting of the `missingValues` property. When you set this property to `true`, DataWeave honors the setting of the `missingValues property`. Introduced in DataWeave 4.3 (2.3.0-20210823) for the September 2021 release of Mule 4.3.0-20210823.

Author: ```suggestion
|`useMissCharAsDefaultForFill`  |`Boolean`|`false`|By default, the flat file reader and writer use spaces for missing characters and ignore the setting of the `missingValues` property. When you set this property to `true`, DataWeave honors the `missingValues property` setting. Introduced in DataWeave 2.3 (2.3.0-20210823) for the September 2021 release of Mule 4.3.0-20210823.
```
